### PR TITLE
Add useCioBreadcrumb

### PR DESCRIPTION
**Pull request by mocca102**

### CODE REVIEW

File: spec/hooks/useCioBreadcrumb/useCioBreadcrumb.test.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        },
+        {
+          display_name: 'Shoes',
+          group_id: 'men-shoes',
+        },
+      ],
+    },
+  ];
+  const filterValue = 'men-shoes-boots';
+  const useBreadCrumbProps = { groups, filterValue };
```
Comment (esezen) [inline spec/hooks/useCioBreadcrumb/useCioBreadcrumb.test.js:51]:
```suggestion
  const useBreadcrumbProps = { groups, filterValue };
```

File: spec/hooks/useCioBreadcrumb/useCioBreadcrumb.test.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+  it('Should return crumbs array', async () => {
+    const { result } = renderHookWithCioPlp(() => useCioBreadcrumb(useBreadCrumbProps));
```
Comment (esezen) [inline spec/hooks/useCioBreadcrumb/useCioBreadcrumb.test.js:54]:
```suggestion
    const { result } = renderHookWithCioPlp(() => useCioBreadcrumb(useBreadcrumbProps));
```

**Review by esezen — APPROVED** [review]
LGTM!

File: src/hooks/useCioBreadcrumb.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
+import { Group } from '@constructor-io/constructorio-client-javascript/lib/types';
+import { useMemo } from 'react';
+
+const generateBreadcrumbs = (groups: Partial<Group>[], filterValue: string) => {
```
Comment (Mudaafi) [inline src/hooks/useCioBreadcrumb.ts:4]: In my PR with groups, I made a new type `PlpGroup` which is essentially the same thing, but with the fields transformed to camel case to match what were doing with the other fields. I think this is fine for now, I'll note down a story to make it consistent later on.

**Review by Mudaafi — APPROVED** [review]
LGTM, simple is best

**Pull request merged**